### PR TITLE
doc: add documentation generator

### PR DESCRIPTION
**Pull request by jerome-quere**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request.
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates to #1169

### CODE REVIEW

File: internal/docgen/tpl.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package docgen
+
+const tplStr = `
+
+
+{{- define "namespace" -}}
+{{ template "do_not_edit" }}
+# Documentation for scw {{ .Cmd.Namespace }}
```
Comment (remyleone) [inline internal/docgen/tpl.go:8]:
```suggestion
# Documentation for `scw {{ .Cmd.Namespace }}`
```

Comment (remyleone) [reply internal/docgen/tpl.go:8]: Looks like a bad idea as we cannot escape simple backtick in golang in raw litterals :(

File: internal/docgen/docgen.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+func renderNamespace(data *tplNamespace) (string, error) {
+	buffer := bytes.Buffer{}
+	err := newTemplate().ExecuteTemplate(&buffer, "namespace", data)
+	if err != nil {
+		return "", err
+	}
+	str := buffer.String()
+	//str = interactive.UnIndent(str, 2)
```
Comment (jerome-quere) [inline internal/docgen/docgen.go:98]: remove

File: internal/docgen/docgen.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	return str, nil
+}
+
+func newTemplate() *template.Template {
+	tpl := template.New("index")
+	tpl = tpl.Funcs(map[string]interface{}{
+		"unindent": func(indent int, str string) string {
```
Comment (jerome-quere) [inline internal/docgen/docgen.go:105]: Don't think this is used

File: internal/docgen/docgen.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+			return "```"
+		},
+		"map": func(args ...interface{}) map[string]interface{} {
+			res := map[string]interface{}{}
+			for i := 0; i < len(args); i += 2 {
+				res[args[i].(string)] = args[i+1]
+			}
+			return res
+		},
+		"join": func(sep string, slice []string) string {
```
Comment (jerome-quere) [inline internal/docgen/docgen.go:121]: Is this used ?

File: internal/docgen/docgen.go [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+			return strings.Join(slice, sep)
+		},
+		"concat": func(strs ...string) string {
```
Comment (jerome-quere) [inline internal/docgen/docgen.go:124]: Is this used ?

File: internal/interactive/print.go [hunk 6]
```diff
@@ -61,6 +61,17 @@ func Indent(str string, indent int) string {
 	return strings.Join(lines, "\n")
 }
 
+func UnIndent(str string, indent int) string {
```
Comment (jerome-quere) [inline internal/interactive/print.go:64]: I think this can be removed

**Pull request merged**